### PR TITLE
OZ-502: GA workflows to use shared workflow to allow image releases.

### PR DESCRIPTION
**Pull request by enyachoke**
This PR updates the GA workflow to use the shared workflow to allow us to release the image. See https://mekomsolutions.atlassian.net/browse/OZ-502

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,52 +2,18 @@ name: CI
...
-          platforms: linux/amd64,linux/arm64
-          push: true
-          tags: ${{ steps.meta.outputs.tags }}
+  docker-release:
+    uses: mekomsolutions/shared-github-workflow/.github/workflows/docker-build-publish.yml@main
+    with:
+      download-artifacts: false
+      image-name: "senaite"
+    secrets:
+      DOCKER_HUB_USERNAME: mekomsolutions
```
Comment (kipchubett) [inline .github/workflows/build.yml]: Let's use username from secrets here.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -2,52 +2,18 @@ name: CI
 
 on:
   push:
-    branches: [ main ]
-  workflow_dispatch:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (kipchubett) [inline .github/workflows/build.yml]: Remove on `pull_request`. We don't want to build and publish docker images on PR

**Review by kipchubett — COMMENTED** [review]
Generally LGTM except the build failure. Are any dockerhub org secrets for mekomsolutions.

File: .github/workflows/build.yml (hunk 1 above)
Comment (enyachoke) [reply .github/workflows/build.yml]: I am not sure we have one. We will need @rbuisson  to confirm

Comment (rbuisson) [reply .github/workflows/build.yml]: Those are the secrets vars available at the org level

Screenshot 2024-05-21 at 14 13 35

**Comment by kipchubett** [comment]
This looks ok, however we need to modify [here](https://github.com/mekomsolutions/shared-github-workflow/blob/main/.github/workflows/docker-build-publish.yml#L57-L62) to not fail when it is not a maven project and also accepts "tags" as input variable which should defaults to empty. This means in additional to `latest` and `version` from pom file(If the project is maven), we can add out own tags thru the inputs.

**Review by kipchubett — APPROVED** [review]

**Pull request merged**